controllers/questionController.js

```
@@ -0,0 +1,107 @@
+        if (allQuestions.length < 5 && cleanedSkills.every(skill => !skillSpecificQuestions[skill.toLowerCase()])) {
+            allQuestions.push(...getRandomQuestions(nonTechnicalQuestions, 3));
+        }
+    
+        // Step 4: Add general technical questions if still fewer than 5 questions
+        if (allQuestions.length < 5) {
+            allQuestions.push(...getRandomQuestions(generalTechnicalQuestions, 3));
+        }
+    
+        // Step 5: Limit to a maximum of 10 questions
+        if (allQuestions.length > 10) {
+            allQuestions = allQuestions.slice(0, 10);
```

Biome

[error] Can't assign allQuestions because it's a constant

This is where the variable is defined as constant

Unsafe fix: Replace const with let if you assign it to a new value.


(lint/correctness/noConstAssign)